Microsoft.Toolkit.Uwp.UI/Extensions/DependencyObjectExtensions.cs

```diff
@@ -460,5 +900,73 @@ public static IEnumerable<DependencyObject> FindAscendants(this DependencyObject
                 element = parent;
             }
         }
+
+        /// <summary>
+        /// Find all ascendant elements of the specified element (or self). This method can be chained
+        /// with LINQ calls to add additional filters or projections on top of the returned results.
+        /// <para>
+        /// This method is meant to provide extra flexibility in specific scenarios and it should not
+        /// be used when only the first item is being looked for. In those cases, use one of the
+        /// available <see cref="FindAscendantOrSelf{T}(DependencyObject)"/> overloads instead,
+        /// which will offer a more compact syntax as well as better performance in those cases.
+        /// </para>
+        /// </summary>
+        /// <param name="element">The root element.</param>
+        /// <returns>All the descendant <see cref="DependencyObject"/> instance from <paramref name="element"/>.</returns>
+        public static IEnumerable<DependencyObject> FindAscendantsOrSelf(this DependencyObject element)
+        {
+            yield return element;
+
+            while (true)
+            {
+                DependencyObject? parent = VisualTreeHelper.GetParent(element);
+
+                if (parent is null)
+                {
+                    yield break;
+                }
+
+                yield return parent;
+
+                element = parent;
+            }
+        }
+
+        /// <summary>
+        /// Checks whether or not a given <see cref="DependencyObject"/> instance is a descendant of another one.
+        /// </summary>
+        /// <param name="child">The input child element.</param>
+        /// <param name="element">The element to look for in the ascendants hierarchy for <paramref name="child"/>.</param>
+        /// <returns>Whether or not <paramref name="child"/> is a descendant of <paramref name="element"/>.</returns>
+        public static bool IsDescendantOf(this DependencyObject child, DependencyObject element)
+        {
+            while (true)
+            {
+                DependencyObject? parent = VisualTreeHelper.GetParent(child);
+
+                if (parent is null)
+                {
+                    return false;
+                }
+
+                if (parent == element)
```

Comment:
Maybe do a ReferenceEquals instead of using the default equality operator?

Reply:
Same thing, `DependencyObject` doesn't override the `==` operator anyway.
This way it's more compact - explicitly using `ReferenceEquals` looks a bit awkward.

Reply:
Threw together a quick test to be sure. Casting changes which equality operator is used, so this is good 👍
![image](https://user-images.githubusercontent.com/9384894/146819800-ec66455c-9f16-4a63-98b3-bbcdf31fa68b.png)